src/scss/foundations/colors.scss

```diff
@@ -0,0 +1,119 @@
+@use "sass:color";
+@use "../lib";
+
+$_colorActionBase: #067f96;
+$_colorActionDark: #e5f1f4;
+$_colorActionLight: #045767;
+
+$_colorAlertBase: #ffdc50;
+$_colorAlertDark: #fff8dc;
+$_colorAlertLight: #ffc010;
+
+$_colorBrandBase: #8334c2;
+$_colorBrandDark: #f2eaf8;
+$_colorBrandLight: #5b2488;
+
+$_colorErrorBase: #df1654;
+$_colorErrorDark: #fbe7ed;
+$_colorErrorLight: #b21143;
+
+$_colorNeutral25: #595d6a;
+$_colorNeutral50: #90939d;
+$_colorNeutral75: #c7c9cd;
+$_colorNeutral85: #dddee1;
+$_colorNeutral95: #f3f4f4;
+$_colorNeutralBase: #21283b;
+$_colorNeutralLight: #ffffff;
+
+$_colorSuccessBase: #1d8147;
+$_colorSuccessDark: #e8f2ec;
+$_colorSuccessLight: #145a31;
+
+/*
+The color palette of Pyxis
+
+@type Map
+
+@access private
+*/
+
+$_colors: (
+  action: (
+    base: $_colorActionBase,
+    dark: $_colorActionDark,
+    light: $_colorActionLight,
+  ),
+  alert: (
+    base: $_colorAlertBase,
+    dark: $_colorAlertDark,
+    light: $_colorAlertLight,
+  ),
+  brand: (
+    base: $_colorBrandBase,
+    dark: $_colorBrandDark,
+    light: $_colorBrandLight,
+  ),
+  error: (
+    base: $_colorErrorBase,
+    dark: $_colorErrorDark,
+    light: $_colorErrorLight,
+  ),
+  neutral: (
+    25: $_colorNeutral25,
+    50: $_colorNeutral50,
+    75: $_colorNeutral75,
+    85: $_colorNeutral85,
+    95: $_colorNeutral95,
+    base: $_colorNeutralBase,
+    light: $_colorNeutralLight,
+  ),
+  success: (
+    base: $_colorSuccessBase,
+    dark: $_colorSuccessDark,
+    light: $_colorSuccessLight,
+  ),
+);
```

Comment:
Secondo me la stiamo ancora gestendo come libreria e non come DS. Forse si. potrebbe definire una mappa di colori sempre validi e di background con i relativi colori che si possono utilizzare. Lasciarla cosi mi ricorda molto bootstrap. Lascio scegliere agli altri quali sono le combinazioni che accetto

Reply:
È una giusta osservazione, ne abbiamo parlato molto anche in fase di design ma nella pratica il discorso è molto più ampio di così. Sarebbe bello avere dei colori definiti solo per il testo o il background (anche se background può voler dire: background-color, border-color, border-color, fill, ecc...) ma in realtà praticamente tutti i colori possono funzionare in un modo o in un altro, a patto che l'accessibilità sia rispettata ([qui](https://contrast-grid.eightshapes.com/?version=1.1.0&background-colors=&foreground-colors=%23067f96%2C%20colorActionBase%0D%0A%23e5f1f4%2C%20colorActionDark%0D%0A%23045767%2C%20colorActionLight%0D%0A%0D%0A%23ffdc50%2C%20colorAlertBase%0D%0A%23fff8dc%2C%20colorAlertDark%0D%0A%23ffc010%2C%20colorAlertLight%0D%0A%0D%0A%238334c2%2C%20colorBrandBase%0D%0A%23f2eaf8%2C%20colorBrandDark%0D%0A%235b2488%2C%20colorBrandLight%0D%0A%0D%0A%23df1654%2C%20colorErrorBase%0D%0A%23fbe7ed%2C%20colorErrorDark%0D%0A%23b21143%2C%20colorErrorLight%0D%0A%0D%0A%23595d6a%2C%20colorNeutral25%0D%0A%2390939d%2C%20colorNeutral50%0D%0A%23c7c9cd%2C%20colorNeutral75%0D%0A%23dddee1%2C%20colorNeutral85%0D%0A%23f3f4f4%2C%20colorNeutral95%0D%0A%2321283b%2C%20colorNeutralBase%0D%0A%23ffffff%2C%20colorNeutralLight%0D%0A%0D%0A%231d8147%2C%20colorSuccessBase%0D%0A%23e8f2ec%2C%20colorSuccessDark%0D%0A%23145a31%2C%20colorSuccessLight&es-color-form__tile-size=compact&es-color-form__show-contrast=aaa&es-color-form__show-contrast=aa) la tabella di accessibilità dei colori, come vedete le varianti possibili sono davvero tante e riportare solo gli accoppiamenti dei colori utilizzati sarebbe più confusionario che altro).

Per questo motivo preferisco avere delle funzioni generiche per i colori (nei foundations almeno) per poi andare a specificare l'utilizzo nei componenti.

p.s.
Poi comunque dalla varie ricerche fatte praticamente tutti i design system (tranne uno: comet) utilizzano variabili o funzioni generiche per i colori, proprio perché poi vanno a definire l'utilizzo all'interno del componente.